bin/cloudgov/deploy-api-proxy

```diff
@@ -0,0 +1,134 @@
+SPACE=$(cf target | awk '/space:/ {print $2}')
+APP="api-proxy"
+
+cf unset-env $APP HTTPS_PROXY
+cf unset-env $APP HTTP_PROXY
+cf unset-env $APP https_proxy
+cf unset-env $APP http_proxy
+
+# We will use $ORG and $SPACE to get unique host names. Sandbox space names have dots, so remove those.
+SPACE_NODOT=${SPACE//\./-}
+
+echo "✅ Deploying $APP..."
+
+# Deploy key storage service
+{
+  echo "🔑 Configuring Key Storage Service."
+  if service_exists "key-storage" ; then
+    echo "✅ Storage already created."
+  else
+    yes '' | cf create-user-provided-service key-storage -p "{}"
+  fi
+}
+
+# Deploy API Proxy
+echo "🚀 Deploying API Proxy..."
+NO_ROUTE="--no-route"
+if appExists $APP; then
+    NO_ROUTE=""
+fi
+
+if cf push -f .buildpack/manifest-$APP.yml ${NO_ROUTE}; then
+  echo "✅ API Proxy deployed successfully."
+else
+  echo "❌ API Proxy deployment failed."
+  exit 1
+fi
+
+# Bind key storage service to the API Proxy
+echo "🔗 Binding Key Storage Service to API Proxy..."
+if cf bind-service $APP key-storage > /dev/null 2>&1; then
+  cf restart $APP
+  echo "✅ Key Storage Service bound successfully."
+else
+  echo "❌ Failed to bind Key Storage Service to API Proxy."
+  exit 1
+fi
+
+# Function to check and create missing routes
+declare -a new_routes=()
+
+function ensure_route {
+  local domain="$1"
+  if cf routes | awk '{print $2, $3}' | grep -q "^$APP-${SPACE_NODOT} ${domain}$"; then
+    return 0  # Route already exists, no need to output anything
+  else
+    cf create-route "${domain}" --hostname "$APP-${SPACE_NODOT}"
+    cf map-route $APP "${domain}" --hostname "$APP-${SPACE_NODOT}"
+    new_routes+=("$APP-${SPACE_NODOT}.${domain}")
+  fi
+}
+
+# Ensure routes exist before deployment
+echo "🔄 Checking routes..."
+ensure_route "apps.internal"
+if [[ "$ORG" == "gsa-tts-usagov" ]]; then
+  ensure_route "usa.gov"
+  ensure_route "app.cloud.gov" # TODO: This is temporary for testing purposes, remove for production
+else
+  ensure_route "app.cloud.gov"
+fi
+
+# If any routes were created, display them in a single message
+if [[ ${#new_routes[@]} -gt 0 ]]; then
+  echo "🌐 Created routes: ${new_routes[*]}"
+else
+  echo "✅ All required routes already exist."
+fi
+
+# Remove default route
+if cf routes | awk '{print $2, $3}' | grep -q "^api-proxy app.cloud.gov$"; then
+  echo "🚫 Removing default route..."
+  yes '' | cf delete-route app.cloud.gov --hostname api-proxy
+fi
+
+# Add network policies
+echo "🔒 Configuring network policies..."
+if cf add-network-policy $APP cms --protocol tcp --port 61443 > /dev/null 2>&1 \
+   && cf add-network-policy cms $APP --protocol tcp --port 61443 > /dev/null 2>&1; then
+  echo "✅ Network policies added."
+else
+  echo "❌ Failed to configure network policies."
+  exit 1
+fi
+
+# Bind shared egress group
+echo "🔗 Binding $APP to shared egress..."
+$SCRIPT_DIR/setup-egress-for-apps $APP
+
+PROXYROUTE=$(cf env $app | grep PROXYROUTE | awk '{print $2}')
+if [ -n "$PROXYROUTE" ]; then
```

Comment:
These need to happen within the app's code, not on the app itself.  The thing that could (and will) go wrong with this approach is that when the app is restaged for any reason, it needs to have these http_proxy variables un-set, and that could happen for reasons other than deployment. 

(I am actually unsure whether "restage" or "restart" is closer to what happens when cloud.gov pushes out an update to their Cloud Foundry platform.) 

You can test this by running "cf restage api-proxy" and watch it fail to install the libraries it needs because of the proxy. 

I think you can do this in a `.profile` file right in the `api-proxy` directory.